### PR TITLE
Forego grace period and force proxy to be set

### PR DESCRIPTION
**Pull request by geowa4**
Due to timelines for requiring the proxy, we're going to require the proxy configuration for AWS immediately.

[OSD-15754](https://issues.redhat.com//browse/OSD-15754)

### CODE REVIEW

File: pkg/provider/aws/client.go [hunk 1]
```diff
@@ -140,15 +139,13 @@ type AwsClient struct {
...
-		_, _ = fmt.Fprintf(os.Stderr, "[WARNING] `%s` not configured. Please add this to your osdctl configuration to ensure traffic is routed though a proxy.\n", ProxyConfigKey)
-		if skipAwsProxyCheck := viper.GetBool(SkipProxyCheckKey); !skipAwsProxyCheck {
-			_, _ = fmt.Fprint(os.Stderr, "Please confirm that you would like to continue with [y|N] ")
-			var input string
-			_, _ = fmt.Scanln(&input)
-			if strings.ToLower(input) != "y" {
-				_, _ = fmt.Fprintln(os.Stderr, "Must enter 'y' to continue; exiting...")
-				os.Exit(0)
-			}
+		_, _ = fmt.Fprintf(os.Stderr, "[ERROR] `%s` not configured. Please add this to your osdctl configuration to ensure traffic is routed though a proxy.\n", ProxyConfigKey)
```
Comment (AlexVulaj) [inline pkg/provider/aws/client.go:142]: This still feels more like a warning to me, as we aren't crashing or hard-enforcing anything in the absence of an `aws_proxy`. I'm not fully opposed, but what made you up the severity here?

File: cmd/account/mgmt/account-unassign_test.go [hunk 2]
```diff
@@ -21,8 +21,7 @@ import (
 )
 
 func TestAssumeRoleForAccount(t *testing.T) {
-	viper.Set(awsInternal.ProxyConfigKey, "")
-	viper.Set(awsInternal.SkipProxyCheckKey, true)
+	viper.Set(awsInternal.ProxyConfigKey, "a-random-proxy")
```
Comment (AlexVulaj) [inline cmd/account/mgmt/account-unassign_test.go:24]: Is this left over from testing? Either way I think we should default to empty-string still.

File: pkg/provider/aws/client.go (hunk 1 above)
Comment (geowa4) [reply pkg/provider/aws/client.go:142]: In my head, warnings are often allowed to accumulate. Errors must be dealt with, which I'm demanding the operator does in the next couple lines of code. But I also don't really mind either way.

File: cmd/account/mgmt/account-unassign_test.go (hunk 2 above)
Comment (AlexVulaj) [reply cmd/account/mgmt/account-unassign_test.go:24]: _sigh_ That's my bad. I totally missed that this was the test file. Carry on.

**Pull request merged**